### PR TITLE
MODORDERS-175 Add ability to move a received piece back to "Expected"

### PR DESCRIPTION
**Pull request by piotr-kalashuk**
## Purpose
[MODORDERS-175](https://issues.folio.org/browse/MODORDERS-175) Add ability to move a received piece back to "Expected"
## Approach
The logic extends functionality implemented in scope of PR #107:
* client sends request like following:
  ```json
  {
    "toBeReceived": [
      {
        "poLineId": "f217a5c2-2c56-4d05-9412-a96cfc8e52de",
        "received": 2,
        "receivedItems": [
          {
            "itemStatus": "On order",
            "pieceId": "56fbfde4-6335-4dd7-9a03-d100821f1d18"
          },
          {
            "itemStatus": "On order",
            "pieceId": "a6ff14ad-8a65-57f9-bb89-f4654a3923fb"
          }
        ]
      }
    ],
    "totalRecords": 2
  }
  ```
* retrieve piece records
* for those piece records which have an item id reference, retrieve items from inventory storage
  * ~~remove item's barcode~~ the item's barcode remains as is
  * update item's status to `On order`
  * update item record in inventory
* update each piece record removing received date and setting receiving status to `Expected`
* retrieve PO Line and calculate receipt status based on quantity of received and expected piece records. If the receipt status was `Fully Received` but the calculated one is either `Awaiting Receipt` or `Partially Received`, the receipt date is removed

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards - [sonarcloud PR#110](https://sonarcloud.io/project/issues?id=org.folio%3Amod-orders&pullRequest=110&resolved=false)?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,12 +75,68 @@ Upon receiving a request to receive resources, it does the following:
...
+              "itemStatus": "On order",
+              "pieceId": "56fbfde4-6335-4dd7-9a03-d100821f1d18"
+            }
+          ]
+        }
+      ],
+      "totalRecords": 1
+    }
+    ```
+
```
Comment (craigmcnally) [inline README.md]: +1 for documentation

File: src/main/java/org/folio/rest/impl/InventoryHelper.java [hunk 2]
```diff
@@ -173,11 +173,17 @@
     itemRecord.put(ITEM_STATUS, new JsonObject().put(ITEM_STATUS_NAME, receivedItem.getItemStatus()));
     if (StringUtils.isNotEmpty(receivedItem.getBarcode())) {
       itemRecord.put(ITEM_BARCODE, receivedItem.getBarcode());
+    } else if (isOnOrderItemStatus(receivedItem)) {
+      itemRecord.remove(ITEM_BARCODE);
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/InventoryHelper.java]: We don't want to remove the barcode, just change the status

File: src/main/java/org/folio/rest/impl/ReceivingHelper.java [hunk 3]
```diff
@@ -239,6 +240,16 @@ private void addPieceIfValid(Piece piece, Map<String, List<Piece>> piecesByPoLin
     }
   }
 
+  /**
+   * Verifies if the current status of the piece record is "Received" and the client would like to roll-back to Expected
+   * @param piece piece reqcord involved in receiving flow
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/ReceivingHelper.java]: typo reqcord

File: src/main/java/org/folio/rest/impl/InventoryHelper.java (hunk 2 above)
Comment (piotr-kalashuk) [reply src/main/java/org/folio/rest/impl/InventoryHelper.java]: Corrected

File: src/main/java/org/folio/rest/impl/ReceivingHelper.java (hunk 3 above)
Comment (piotr-kalashuk) [reply src/main/java/org/folio/rest/impl/ReceivingHelper.java]: Corrected

**Pull request merged**